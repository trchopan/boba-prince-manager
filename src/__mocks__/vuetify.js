export const mockVuetifyComponents = [
  "v-app",
  "v-toolbar-items",
  "v-toolbar-side-icon",
  "v-navigation-drawer",
  "v-content",
  "v-container",
  "v-layout",
  "v-spacer",
  "v-flex",
  "v-btn",
  "v-list",
  "v-card",
  "v-card-text",
  "v-card-actions",
  "v-toolbar",
  "v-toolbar-title",
  "v-card-text",
  "v-form",
  "v-progress-linear",
  "v-text-field",
  "v-list-tile",
  "v-list-tile-title",
  "v-divider",
  "v-icon",
  "v-list-tile-avatar",

];
