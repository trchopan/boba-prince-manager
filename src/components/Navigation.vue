<template>
  <v-app>
    <v-toolbar app flat clipped-left dark>
      <v-toolbar-side-icon @click="drawerOpen = !drawerOpen"></v-toolbar-side-icon>
      <v-toolbar-title>{{ title }}</v-toolbar-title>
      <v-spacer></v-spacer>
      <v-toolbar-items class="hidden-sm-and-down">
        <v-btn flat>Link One</v-btn>
        <v-btn flat>Link Two</v-btn>
        <v-btn flat>Link Three</v-btn>
      </v-toolbar-items>
    </v-toolbar>
    <v-navigation-drawer
      app
      clipped
      dark
      v-model="drawerOpen">
        <v-list class="pa-0">
          <v-list-tile avatar>
            <v-list-tile-avatar>
              <img src="https://randomuser.me/api/portraits/men/85.jpg">
            </v-list-tile-avatar>
            <v-list-tile-content>
              <v-list-tile-title>John Leider</v-list-tile-title>
            </v-list-tile-content>
          </v-list-tile>
          <v-list-tile @click="logout()">
            <v-list-tile-action>
              <div>&nbsp;</div>
            </v-list-tile-action>
            <v-list-tile-content>
              <v-list-tile-title>Log out</v-list-tile-title>
            </v-list-tile-content>
          </v-list-tile>
        </v-list>
    </v-navigation-drawer>
    <v-content>
      <v-container fluid>
        <slot></slot>
      </v-container>
    </v-content>
  </v-app>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  name: "Navigation",
  data() {
    return {
      drawerOpen: false,
      bottomNav: "recent",
      title: process.env.VUE_APP_TITLE
    };
  },
  computed: mapGetters({ currentUser: "auth/currentUser" }),
  methods: {
    async logout() {
      const loggedOut = await this.$store.dispatch("auth/logout");
      if (loggedOut) this.$router.replace("/login");
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
