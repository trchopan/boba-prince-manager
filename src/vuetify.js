import Vue from "vue";
import Vuetify from "vuetify/lib";
import colors from "vuetify/es5/util/colors";

import {
  VApp,
  VAvatar,
  VContainer,
  VContent,
  VLayout,
  VFlex,
  VCard,
  VCardText,
  VCardActions,
  VDivider,
  VForm,
  VTextField,
  VTooltip,
  VToolbar,
  VToolbarSideIcon,
  VToolbarTitle,
  VToolbarItems,
  VMenu,
  VMenuTransition,
  VNavigationDrawer,
  VList,
  VListTile,
  VListTileAvatar,
  VListTileContent,
  VListTileTitle,
  VListTileSubTitle,
  VListTileAction,
  VProgressLinear,
  VProgressCircular,
  VRating,
  VBtn,
  VFabTransition,
  VIcon,
  VSwitch,
  VSelect,
  VSpacer
  // transitions
} from "vuetify/lib/components";
import { Ripple } from "vuetify/lib/directives";
import "vuetify/src/stylus/app.styl";

const components = {
  VApp,
  VAvatar,
  VContainer,
  VContent,
  VLayout,
  VFlex,
  VCard,
  VCardText,
  VCardActions,
  VDivider,
  VForm,
  VTextField,
  VTooltip,
  VToolbar,
  VToolbarSideIcon,
  VToolbarTitle,
  VToolbarItems,
  VMenu,
  VMenuTransition,
  VNavigationDrawer,
  VList,
  VListTile,
  VListTileAvatar,
  VListTileContent,
  VListTileTitle,
  VListTileSubTitle,
  VListTileAction,
  VProgressLinear,
  VProgressCircular,
  VRating,
  VBtn,
  VFabTransition,
  VIcon,
  VSwitch,
  VSelect,
  VSpacer
  // transitions
};
const directives = {
  Ripple
};

Vue.use(Vuetify, {
  components,
  directives,
  theme: {
    secondary: colors.pink.accent2,
    accent: colors.indigo.base,
    warn: colors.red.accent2
  }
});
