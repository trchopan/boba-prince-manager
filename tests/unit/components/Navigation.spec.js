import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Navigation from "@/components/Navigation.vue";
import { mockVuetifyComponents } from "@/__mocks__/vuetify.js";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("components/Navigation.vue", () => {
  let store;
  let getters;
  let actions;
  let wrapper;
  let $vuetify;

  beforeEach(() => {
    getters = {
      "auth/authUser": () => null,
      "auth/profile": () => null
    };
    actions = {
      "auth/logout": jest
        .fn()
        .mockImplementationOnce(() => Promise.resolve(false))
        .mockImplementationOnce(() => Promise.resolve(true))
    };
    store = new Vuex.Store({
      getters,
      actions
    });
    $vuetify = {
      breakpoint: { name: "lg", mdAndUp: true, lgAndUp: true }
    };
    wrapper = shallowMount(Navigation, {
      stubs: mockVuetifyComponents,
      mocks: { $router: { push: jest.fn(), replace: jest.fn() }, $vuetify },
      store,
      localVue
    });
  });
  it("logouts and handle error", async () => {
    await wrapper.vm.logout();
    expect(actions["auth/logout"]).toBeCalled();
    expect(wrapper.vm.$router.replace).not.toBeCalled();

    await wrapper.vm.logout();
    expect(wrapper.vm.$router.replace.mock.calls).toEqual([["/login"]]);
  });
  it("navigates", () => {
    let link = "/link";
    wrapper.vm.navigate(link);
    expect(wrapper.vm.$router.push.mock.calls).toEqual([[link]]);
  })
  it("matchs snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
