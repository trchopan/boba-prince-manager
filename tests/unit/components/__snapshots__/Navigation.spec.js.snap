// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation.vue matchs snapshot 1`] = `
<v-app-stub>
  <v-toolbar-stub app="" flat="" clipped-left="" dark="">
    <v-toolbar-side-icon-stub></v-toolbar-side-icon-stub>
    <v-toolbar-title-stub>Boba Prince ERP</v-toolbar-title-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-toolbar-items-stub>
      <v-btn-stub flat="">
        Log in
      </v-btn-stub>
    </v-toolbar-items-stub>
  </v-toolbar-stub>
  <v-navigation-drawer-stub app="" dark="" clipped="" width="200">
    <v-list-stub class="pa-0">
      <v-list-tile-stub avatar="">
        <v-list-tile-avatar-stub>
          <v-icon-stub>home</v-icon-stub>
        </v-list-tile-avatar-stub>
        <v-list-tile-title-stub>Home</v-list-tile-title-stub>
      </v-list-tile-stub>
      <v-list-tile-stub avatar="">
        <v-list-tile-avatar-stub>
          <v-icon-stub>dashboard</v-icon-stub>
        </v-list-tile-avatar-stub>
        <v-list-tile-title-stub>Dashboard</v-list-tile-title-stub>
      </v-list-tile-stub>
      <v-list-tile-stub avatar="">
        <v-list-tile-avatar-stub>
          <v-icon-stub></v-icon-stub>
        </v-list-tile-avatar-stub>
        <v-list-tile-title-stub>About</v-list-tile-title-stub>
      </v-list-tile-stub>
      <v-divider-stub dark=""></v-divider-stub>
    </v-list-stub>
  </v-navigation-drawer-stub>
  <v-content-stub>
    <v-container-stub fluid=""></v-container-stub>
  </v-content-stub>
</v-app-stub>
`;
